src/AppInstallerCLICore/Workflows/ShellExecuteInstallerHandler.cpp

```diff
@@ -247,4 +267,82 @@ namespace AppInstaller::CLI::Workflow
         installerPath.assign(renamedDownloadedInstaller);
         AICLI_LOG(CLI, Info, << "Successfully renamed downloaded installer. Path: " << installerPath);
     }
+
+    void ShellExecuteUninstallImpl(Execution::Context& context)
+    {
+        context.Reporter.Info() << Resource::String::UninstallFlowStartingPackageUninstall << std::endl;
+        std::wstring commandUtf16 = Utility::ConvertToUTF16(context.Get<Execution::Data::UninstallString>());
+
+        // Parse the command string as application and command line for CreateProcess
+        wil::unique_cotaskmem_string app = nullptr;
+        wil::unique_cotaskmem_string args = nullptr;
+        THROW_IF_FAILED(SHEvaluateSystemCommandTemplate(commandUtf16.c_str(), &app, NULL, &args));
+
+        auto uninstallResult = context.Reporter.ExecuteWithProgress(
+            std::bind(InvokeShellExecute,
+                std::filesystem::path(app.get()),
+                Utility::ConvertToUTF8(args.get()),
+                std::placeholders::_1));
+
+        if (!uninstallResult)
+        {
+            context.Reporter.Warn() << Resource::String::UninstallAbandoned << std::endl;
+            AICLI_TERMINATE_CONTEXT(E_ABORT);
+        }
+        else if (uninstallResult.value() != 0)
+        {
+            const auto installedPackageVersion = context.Get<Execution::Data::InstalledPackageVersion>();
+            Logging::Telemetry().LogUninstallerFailure(
+                installedPackageVersion->GetProperty(PackageVersionProperty::Id),
+                installedPackageVersion->GetProperty(PackageVersionProperty::Version),
+                "UninstallString",

```

Comment:
>"UninstallString" [](start = 16, length = 17)

I think this is for InstallerType in telemetry backend? Or is it intentional you create a new type for this telemetry event? Same below for line 338

Reply:
I copied the telemetry event from the installer failure one. I kept the type string so we can distinguish when a failure comes from an Exe and when it is from and Msi as they will likely have different errors.